File: Jessica/20.Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> S;
        for(int i=0;i<s.length();i++){
            if(s[i]=='(' || s[i]=='[' ||s[i]=='{' ){
                cout<<s[i]<<endl;
                S.push(s[i]);
            }         
            else if(!S.empty()){
                char v=S.top();
                if(s[i]==')' && v=='('){
                    S.pop();
                    continue;
                }
                   
                else if(s[i]==']' && v=='['){
                    S.pop();
                    continue;
                }
                 
                else  if(s[i]=='}' && v=='{'){
                    S.pop();
                    continue;
                }
                else
                    return false;
            }
            else
                return false;
        }
        if(S.empty())
            return true;
        else
            return false;
    }
};

File: Jessica/20.Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s)%2==1:
            return False
        stack=[]
        flag=0
        for i in s:
            if i=='(' or i=='[' or i=='{':
                stack.append(i)
            elif i==')' and len(stack)>0 and stack[-1]=='(':
                stack=stack[:-1]
            elif i==']' and len(stack)>0 and stack[-1]=='[':
                stack=stack[:-1]
            elif i=='}' and len(stack)>0 and stack[-1]=='{':
                stack=stack[:-1]
            else:                
                return False
        
        if len(stack)==0:
            return True
        else:
            return False
